Update Dependabot config

Pl217 opened the pull request.

Pl217 committed:
Remove unnecessary `dependabot.yml` configuration

Don't be explicit about package and their `@types/`
variant in `dependabot.yml`.

It was proven in practice that Dependabot packs
these together even without the config. Group
named "jest" is dept because it has third package
called `ts-jest`, which is unclear whether Dependabot
can pick up automatically

Pl217 committed:
Allow Dependabot to bump versions of ESLint and Typescript

Pl217 committed:
Update `dependabot.yml` documentation link

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 committed:
Remove `reviewers` from `dependabot.yml`

The `reviewers` field in the `dependabot.yml`
file will be removed soon. Please use the code
owners file to specify reviewers for Dependabot PRs.

For more information, see this blog post
https://github.blog/changelog/2025-04-29-dependabot-reviewers-configuration-option-being-replaced-by-code-owners

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review
